src/lib/optimizer/strategy/between_composition_rule.hpp

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "abstract_rule.hpp"
+#include "expression/expression_functional.hpp"
+
+#include "types.hpp"
+
+namespace opossum {
+
+class AbstractLQPNode;
+class PredicateNode;
+
+/**
+ * Rewrite "column_a >= x AND column_a <= y" to "column_a BETWEEN x AND y" for a performance boost.
+ *
+ * The BetweenCompositionRule searches for a chain of predicate nodes and substitutes BinaryPredicateConditions
+ * to BetweenExpressions within this chain. The algorithm checks whether two or more BinaryPredicateConditions
+ * represent a range on one column. The highest lower bound and the lowest upper bound are substituted by a
+ * corresponding (exclusive or inclusive) BetweenExpression. All obsolete BinaryPredicateConditions are removed
+ * after the substitution.
+**/
+class BetweenCompositionRule : public AbstractRule {
+ public:
+  std::string name() const override;
+  void apply_to(const std::shared_ptr<AbstractLQPNode>& node) const override;
+
+ private:
+  /**
+ * The ColumnBoundaryType defines whether a value represents a boundary for a column or not (NONE)
+ * and if it is a boundary it also defines which kind of boundary it is including the inclusive and exclusive property.
+**/
+  enum class ColumnBoundaryType {
+    None,
+    LowerBoundaryInclusive,
+    LowerBoundaryExclusive,
+    UpperBoundaryInclusive,
+    UpperBoundaryExclusive,
+  };
+
+  /**
```

Comment:
did you run the formatter? These comments usually have their *s aligned.

Reply:
yes, but maybe the IDE directly indented it again.

somehow, the linter did not complain, but I think it should in such situations? Our working CI is a proof for the linter not checking this.

Should this be a new issue? As I am running out of time I do not want it to be in our own scope, sorry.